infrastructure/templates/public-api/main.bicep

```diff
@@ -116,68 +118,137 @@ module fileShareModule 'components/fileShares.bicep' = {
   }
 }
 
+// In order to link PostgreSQL Flexible Server to a VNet, it must have a Private DNS zone available with a name ending
+// with "postgres.database.azure.com".
+resource postgreSqlPrivateDnsZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
+  name: 'private.postgres.database.azure.com'
+  location: 'global'
+  resource vNetLink 'virtualNetworkLinks' = {
+    name: '${resourcePrefix}-psql-flexibleserver-vnet-link'
+    location: 'global'
+    properties: {
+      registrationEnabled: false
+      virtualNetwork: {
+        id: vNetModule.outputs.vNetRef
+      }
+    }
+  }
+}
+
 // Deploy PostgreSQL Database.
 module postgreSqlServerModule 'components/postgresqlDatabase.bicep' = {
   name: 'postgreSQLDatabaseDeploy'
   params: {
     resourcePrefix: resourcePrefix
     location: location
+    createMode: psqlDbUsersAdded ? 'Update' : 'Default'
     adminName: postgreSqlAdminName!
     adminPassword: postgreSqlAdminPassword!
     dbSkuName: postgreSqlSkuName
     dbStorageSizeGB: postgreSqlStorageSizeGB
     dbAutoGrowStatus: postgreSqlAutoGrowStatus
-    keyVaultName: keyVaultName
+    postgreSqlVersion: '16'
     tagValues: tagValues
-    vNetId: vNetModule.outputs.vNetRef
+    privateDnsZoneId: postgreSqlPrivateDnsZone.id
     firewallRules: postgreSqlFirewallRules
     subnetId: vNetModule.outputs.postgreSqlSubnetRef
     databaseNames: ['public_data']
   }
-  dependsOn: [
-    vNetModule
-  ]
+}
+
+resource apiContainerAppManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: apiContainerAppManagedIdentityName
+  location: location
+}
+
+var acrPullRole = resourceId('Microsoft.Authorization/roleDefinitions', '7f951dda-4ed3-4680-a7ca-43fe172d538d')
+
+@description('This allows the managed identity of the container app to access the registry, note scope is applied to the wider ResourceGroup not the ACR')
+resource apiContainerAppManagedIdentityRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid(resourceGroup().id, apiContainerAppManagedIdentity.id, acrPullRole)
+  properties: {
+    roleDefinitionId: acrPullRole
+    principalId: apiContainerAppManagedIdentity.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
+  name: '${resourcePrefix}-log-${apiContainerAppName}'
+  location: location
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+  }
+  tags: tagValues
+}
+
+resource apiContainerAppEnvironment 'Microsoft.App/managedEnvironments@2023-05-01' = {
+  name: '${resourcePrefix}-cae-${apiContainerAppName}'
```

Comment:
I don't think it's really a problem here but the container app environment can be used for more than one container app where the apps are related. Apps would only use separate container app environments where they need to be isolated or shouldn't share the same resources. Should we name the container app environment something more generic than the app itself to reflect this?

Reply:
Hey Ben, yeah that sounds like a good idea.  I've already done this for the PostgreSQL Flexible Server to make it reusable elsewhere, so I'll follow the same convention here, ta

Reply:
Done.  Also done for the Log Analytics Workspace, as I believe that, as we've done with the Public API App Insights instance, it is shareable across several resources and might prove better insights by being combined.